sirepo/feature_config.py

```diff
@@ -109,6 +119,10 @@ def b(msg, dev=False):
         # No secrets should be stored here (see sirepo.job.agent_env)
         api_modules=((), set, 'optional api modules, e.g. status'),
         default_proprietary_sim_types=(set(), set, 'codes where all users are authorized by default but that authorization can be revoked'),
+        dynamic_sim_types=(
+            '',
+            _parse_to_dict,
```

Comment:
@robnagler I have a feeling you won't like this. How should I best supply the cfg?

Reply:
I could play Socrates here  and ask "Why not?" I won't. ;-)

The functional reason this is incorrect is that a single private repo might contain multiple codes (likely) or override parts of different codes (less likely). A simpler search solution would allow us to solve both problems. 

This should just a list of root packages which are checked for existence at bootstrap time that is you should be able to import the root packages themselves. See my comment below about sirepo being implicit or explicit.

Reply:
:)

FWIW `_parse_to_dict` can handle multiple codes from one package `code1:rsprivate:code2:rsprivate`.

How should we get sim_types? We will need to update `_cfg.sim_types` with them and they can't be verified against `VALID_CODES`.

Reply:
The VALID_CODES test probably should be removed or changed. If a code isn't in VALID_CODES then when the code is imported, it will fail. That's ok. If we have a type in sim_types config, that's probably ok, too. Not great, but unlikely on prod.

Reply:
58607069